src/compas/datastructures/mesh/slice.py

```diff
@@ -146,11 +146,7 @@ def intersect(self):
             x = intersection_segment_plane((a, b), self.plane)
             if not x:
                 continue
-            if len(x) == sum([1 for i, j in zip(x, a) if i == j]):
-                intersections.append(u)
-            elif len(x) == sum([1 for i, j in zip(x, b) if i == j]):
-                intersections.append(v)
-            else:
+            if not len(x) == sum([1 for i, j in zip(x, a) if i == j]) and not len(x) == sum([1 for i, j in zip(x, b) if i == j]):
```

Comment:
You want to check whether there is at least one index `n` where `x[n]` is not equal to `a[n]` and similarly for `x` and `b`, right?  I find the following to be an easier-to-read version of that, and also, it should run much faster.

```suggestion
            if any([i != j for i, j in zip(x, a)]) and any([i != j for i, j in zip(x, b)]):
```

but this is only a local comment.  I don't know enough about slice to approve this...

Reply:
Thank you @beverlylytle :) its a bit slower ( 04.549066 sec, instead of 04.491115) , but definitely shorter and easier-to-read

Reply:
@brgcode What do you think of this? thank you